api/envoy/config/core/v3/base.proto

```diff
@@ -323,9 +323,17 @@ message HeaderValueOption {
   // Header name/value pair that this option applies to.
   HeaderValue header = 1 [(validate.rules).message = {required: true}];
 
-  // Should the value be appended? If true (default), the value is appended to
-  // existing values. Otherwise it replaces any existing values.
+  // Should the value be appended? If true (default), the value is appended to the existing values.
+  // It replaces any existing values if `overwrite` field is set to true or otherwise skip any changes.
+  // If both the `append` and `overwrite` are set to false then the given header is only appended
+  // if it doesn't already exist.
   google.protobuf.BoolValue append = 2;
+
+  // Should the value be overwritten? If true (default), the value of existing header is overwritten
+  // when `append` is false.
+  // If both the `append` and `overwrite` are set to false then the given header is only appended
+  // if it doesn't already exist.
+  google.protobuf.BoolValue overwrite = 3;
```

Comment:
**Note to reviewers:**

I couldn't think of any other name here. If there are any other suggestions please let me know and I'll change it accordingly.

Reply:
What would be the semantical meaning if both `append` and `override` were set to true?

Perhaps a better alternative would be an enum type instead of the set of bools. Something like `overrideExisting` (default), `appendToExisting` and `addIfMissing`.

Reply:
or just rename this bool field to `add_if_missing`, then it only controls whether to add new header if there is no one. The enum type is good when we can remove the `append` field, since it looks like duplicated with enum type new field. But we can't remove the `append` for back-forward compatibility I guess.

Reply:
From high level the options are 3 
replace-or-override
append
append-if-exist

I would suggest create new enum field and deprecate the existing `append`

Reply:
+1 to deprecate and add an enum. I think this is confusing. I would just translate the existing append field into the enum and then offer the enum value for the new behavior. Thank you!

/wait

Reply:
Changed it to the `enum` as per the suggestion. Thanks!